src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Range.cs

```diff
@@ -0,0 +1,173 @@
+                    // translates to:       nullableLeft.HasValue ? new Nullable(Range.Create(nullableLeft.GetValueOrDefault(), right)) : default
+
+                    // Case 9:              "left..nullableRight"
+                    // translates to:       nullableRight.HasValue ? new Nullable(Range.Create(left, nullableRight.GetValueOrDefault())) : default
+
+                    MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__Create);
+                    return LiftRangeExpression(node.Syntax, node.Type, method, leftOperand, rightOperand);
+                }
+            }
+        }
+
+        private BoundNode LiftRangeExpression(SyntaxNode syntax, TypeSymbol type, MethodSymbol method, params BoundExpression[] operands)
+        {
+            Debug.Assert(type.IsNullableType());
+            Debug.Assert(operands.Any(operand => operand.Type.IsNullableType()));
+            Debug.Assert(operands.Length == 1 || operands.Length == 2);
+
+            ArrayBuilder<BoundExpression> sideeffects = ArrayBuilder<BoundExpression>.GetInstance();
+            ArrayBuilder<LocalSymbol> locals = ArrayBuilder<LocalSymbol>.GetInstance();
+            ArrayBuilder<BoundExpression> arguments = ArrayBuilder<BoundExpression>.GetInstance();
+
+            // left.HasValue && right.HasValue
+            BoundExpression condition = null;
+            foreach (var operand in operands)
+            {
+                if (operand.Type.IsNullableType())
+                {
+                    BoundExpression tempOperand = CaptureExpressionInTempIfNeeded(operand, sideeffects, locals);
+                    BoundExpression operandHasValue = MakeOptimizedHasValue(tempOperand.Syntax, tempOperand);
+
+                    if (condition is null)
+                    {
+                        condition = operandHasValue;
+                    }
+                    else
+                    {
+                        TypeSymbol boolType = _compilation.GetSpecialType(SpecialType.System_Boolean);
+                        condition = MakeBinaryOperator(syntax, BinaryOperatorKind.BoolAnd, condition, operandHasValue, boolType, method: null);
+                    }
+
+                    arguments.Add(MakeOptimizedGetValueOrDefault(tempOperand.Syntax, tempOperand));
+                }
+                else
+                {
+                    arguments.Add(operand);
+                }
+            }
+
+            Debug.Assert(condition != null);
+
+            // method(left.GetValueOrDefault(), right.GetValueOrDefault())
+            BoundExpression rangeCall = MakeCall(syntax, rewrittenReceiver: null, method, arguments.ToImmutableAndFree(), type);
+
+            // new Nullable(method(left.GetValueOrDefault(), right.GetValueOrDefault()))
+            MethodSymbol nullableCtor = (MethodSymbol)_compilation.GetSpecialTypeMember(SpecialMember.System_Nullable_T__ctor).SymbolAsMember((NamedTypeSymbol)type);
```

Comment:
>SymbolAsMember [](start = 127, length = 14)

Why this is safe to do? I think GetSpecialTypeMember returns null if constructor doesn't exist.

Reply:
I'll add a note to check in semantics just in case.

---
In reply to: [185367467](https://github.com/dotnet/roslyn/pull/26405#discussion_r185367467) [](ancestors = 185367467)

Reply:
> I'll add a note to check in semantics just in case.

I think there are plenty of places in LocalRewriter that do the same thing. Either extract/find a helper that can be reused, or follow the same approach.

---
In reply to: [185370600](https://github.com/dotnet/roslyn/pull/26405#discussion_r185370600) [](ancestors = 185370600,185367467)